package-examples/kustomize-pkg/README.md

```diff
@@ -0,0 +1,163 @@
+├── [Kptfile]  Kptfile kustomize-pkg
+├── PKG: nginx
+│   ├── [Kptfile]  Kptfile nginx
+│   ├── [deployment.yaml]  Deployment my-nginx
+│   └── [svc.yaml]  Service my-nginx-svc
+├── dev
+│   ├── [kustomization.yaml]  Kustomization 
+│   └── [pass-patch.yaml]  Deployment deployment-patch
+└── prod
+    ├── [kustomization.yaml]  Kustomization 
+    └── [pass-patch.yaml]  Deployment deployment-patch
+```
+
+### kustomize the config
+
+We recommend that you keep the kustomize instructions to rendering only such as adding a namespace, transforming or applying a patch.  The second you mix kpt packages and remote bases you will be missing out on a big advantage of having a guaranteed stable base.
+
+You can edit the patches and kustomize files we have created in the overlay folders.
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- ../../bases/nginx
+patches:
+- path: pass-patch.yaml
+  target:
+    kind: Deployment
+commonLabels:
+    environ: dev
+namePrefix: dev-
+```
+
+### Render the kustomization
+
+You can see the final configuration that the patch is going to render. The major difference between this and in place edit is that you can't just go to the file and look at it in isolation, you need to run `kustomize build` and examine the final results.
+
+```shell
+$ kustomize build kustomize-pkg/overlays/dev
+
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: nginx
+    environ: dev
+  name: dev-my-nginx-svc
+spec:
+  ports:
+  - port: 80
+  selector:
+    app: nginx
+    environ: dev
+  type: LoadBalancer
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    environ: dev
+  name: dev-my-nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+      environ: dev
+  template:
+    metadata:
+      labels:
+        app: nginx
+        environ: dev
+    spec:
+      containers:
+      - image: nginx:1.14.1
+        name: nginx
+        ports:
+        - containerPort: 80
+---
+apiVersion: kpt.dev/v1alpha2
+kind: Kptfile
+metadata:
+  labels:
+    environ: dev
+  name: dev-nginx
+pipeline:
+  validators:
+  - image: gcr.io/kpt-fn/kubeval:v0.1
+upstream:
+  git:
+    directory: package-examples/nginx
+    ref: v0.2
+    repo: https://github.com/GoogleContainerTools/kpt
+  type: git
+  updateStrategy: resource-merge
+upstreamLock:
+  git:
+    commit: 4d2aa98b45ddee4b5fa45fbca16f2ff887de9efb
+    directory: package-examples/nginx
+    ref: package-examples/nginx/v0.2
+    repo: https://github.com/GoogleContainerTools/kpt
+  type: git
+```
+
+### Apply the package
+
+It's possible to use kustomize build and kpt live apply, but it does require passing the inventory to `kpt live apply` from `kustomize build` output.  One of the base Kptfiles needs to be initialized with the inventory object.  We will use the nginx base, but in real life exployment it might be better to initialize inventory on your root package.
+
+```shell
+$ kpt live init bases/nginx
```

Comment:
I don't see a `kpt live apply` command below.

Also, is the assumption that the they want to use `kpt live apply`? Or does the user want to use packaging but not hydration and live? Providing both options makes sense.

Reply:
Live apply from STDIN currently doesn't work so that's why we don't have the output.  What I have seen is that people who are using kustomize and see the value in kpt usually like both areas: packaging and apply.  They generally already know what to do with `kubectl -k` or `kustomize build` and piping to `kubectl -f -`.  Providing them an example of doing that probably doesn't add any value in addition to what they can already gather from kustomize docs.